addons/l10n_it_edi_sdicoop/models/account_edi_format.py

```diff
@@ -0,0 +1,267 @@
+
+    def _l10n_it_post_invoices_step_1(self, invoices):
+        ''' Send the invoices to the proxy.
+        '''
+        to_return = {}
+
+        to_send = {}
+        for invoice in invoices:
+            xml = b"<?xml version='1.0' encoding='UTF-8'?>" + invoice._export_as_xml()
+            filename = self._l10n_it_edi_generate_electronic_invoice_filename(invoice)
+            attachment = self.env['ir.attachment'].create({
+                'name': filename,
+                'res_id': invoice.id,
+                'res_model': invoice._name,
+                'datas': base64.encodebytes(xml),
+                'description': _('Italian invoice: %s', invoice.move_type),
+                'type': 'binary',
+            })
+            invoice.l10n_it_edi_attachment_id = attachment
+
+            if len(invoice.commercial_partner_id.l10n_it_pa_index or '') == 6:
+                invoice.message_post(
+                    body=(_("Invoices for PA are not managed by Odoo, you can download the document and send it on your own."))
+                )
+                to_return[invoice] = {'attachment': attachment}
+            else:
+                to_send[filename] = {
+                    'invoice': invoice,
+                    'data': {'filename': filename, 'xml': base64.b64encode(xml)}}
+
+        company = invoices.company_id
+        proxy_user = self._get_proxy_user(company)
+        if not proxy_user:  # proxy user should exist, because there is a check in _check_move_configuration
+            return {invoice: {
+                'error': _("You must accept the terms and conditions in the settings to use FatturaPA."),
+                'blocking_level': 'error'} for invoice in invoices}
+        try:
+            responses = self._l10n_it_edi_upload([i['data'] for i in to_send.values()], proxy_user)
+        except AccountEdiProxyError as e:
+            return {invoice: {'error': e.message, 'blocking_level': 'error'} for invoice in invoices}
+
+        for filename, response in responses.items():
+            invoice = to_send[filename]['invoice']
+            to_return[invoice] = response
+            if 'id_transaction' in response:
+                invoice.l10n_it_edi_transaction = response['id_transaction']
+                to_return[invoice].update({
+                    'error': _('The invoice was successfully transmitted to the Public Administration and we are waiting for confirmation.'),
+                    'blocking_level': 'info',
+                })
+        return to_return
+
+    def _l10n_it_post_invoices_step_2(self, invoices):
+        ''' Check if the sent invoices have been processed by FatturaPA.
+        '''
+        to_check = {i.l10n_it_edi_transaction: i for i in invoices}
+        to_return = {}
+        company = invoices.company_id
+        proxy_user = self._get_proxy_user(company)
+        if not proxy_user:  # proxy user should exist, because there is a check in _check_move_configuration
```

Comment:
I wonder also if this is still relevant...  It could become a AccountEdiProxyError as well...

Reply:
We are not yet in the scope of proxy_user, raising AccountEdiProxyError wouldn't make sense because if we are in a cron, the user won't see the failure. Instead we show a typical account_edi error with blocking_level so that it appears in the banner of the invoice.

Also, as the comment states, this should not happen because there is a check in _check_move_configuration, however, this is defensive programming.